exporter/splunkhecexporter/client.go

```diff
@@ -259,7 +261,22 @@ func (c *client) postEvents(ctx context.Context, events io.Reader, compressed bo
 	io.Copy(ioutil.Discard, resp.Body)
 	resp.Body.Close()
 
-	return splunk.HandleHTTPCode(resp)
+	if err = splunk.HandleHTTPCode(resp); err != nil && resp.StatusCode == http.StatusBadRequest {
+		var err2 error
+
+		if req.Body, err2 = req.GetBody(); err2 != nil {
```

Comment:
Since @bogdandrutu was concerned about exposing PII in the logs let's instead of logging the request log the response. Supposedly the response should describe what exactly was Bad in the request. What do you think?

Also, I think we can do this handling inside splunk.HandleHTTPCode so that it works for all our protocols, not just HEC.

Reply:
Ok, I agree.

Reply:
Can you also update the description of the PR